Ghidra/Features/Base/src/main/java/ghidra/app/util/exporter/BinaryExporter.java

```diff
@@ -39,48 +42,47 @@ public BinaryExporter() {
 
 	@Override
 	public boolean export(File file, DomainObject domainObj, AddressSetView addrSet,
-			TaskMonitor monitor) throws IOException, ExporterException {
+			TaskMonitor monitor) throws IOException {
 
 		if (!(domainObj instanceof Program)) {
 			log.appendMsg("Unsupported type: " + domainObj.getClass().getName());
 			return false;
 		}
-		Program program = (Program) domainObj;
+		final Program program = (Program) domainObj;
+		final Memory memory = program.getMemory();
 
-		Memory memory = program.getMemory();
+		try (final RandomAccessFile fout = new RandomAccessFile(file, "rw")) {
+			final List<FileBytes> fileBytes = memory.getAllFileBytes();
+			if (!fileBytes.isEmpty()) {
+				final FileBytes fByte = fileBytes.get(0);
+				long offset = 0;
+				long size = fByte.getSize();
+				while (size > Integer.MAX_VALUE) {
+					final byte[] bytes = new byte[Integer.MAX_VALUE];
```

Comment:
a 2GB buffer seems a bit extreme, or is that common in other Ghidra code

Reply:
Yes you're right, that is a bit extreme. I actually did that as a "just in case". I'll limit it later tonight.